### PR TITLE
docs: document test-matchers with examples and patterns

### PR DESCRIPTION
**Pull request by alephao**
Related to #536

Created a section in `cli/README.md` (not sure if it's the best place) with examples on how to run a subset of tests. Also added common patterns (that I use) and an edge cases.

Not sure if it's in sync with what other people had in mind, feel free to make suggestions or completely discard it.

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -283,6 +283,95 @@ no files changed, compilation skipped.
...
+```solidity
+contract ContractFoo { ... }
+contract ContractFooBar { ... }
+contract ContractBar { ... }
+```
+
+* `forge test --match-contract Contract` will run the tests in all of those contracts
+* `forge test --match-contract Foo` will run the tests in the `ContractFoo` and `ContractFooBar`
+* `forge test --match-contract "Foo$"` will only run the tests in `ContractFoo`
+* `forge test --match-contract "ContractFoo|ContractBar"` will only run the tests of contracts including `ContractA` or `ContractB` in it's name
```
Comment (gakonst) [inline cli/README.md:312]:
```suggestion
* `forge test --match-contract "ContractFoo|ContractBar"` will only run the tests of contracts including `ContractFoo` or `ContractBar` in it's name
```

**Review by gakonst — APPROVED** [review]

**Review by gakonst — APPROVED** [review]
Great work. Cc @onbjerg for foundry book
